app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,250 @@
+        if c_user.role == Role.admin or c_user.role == Role.moderator
+
+            # return the courses as a JSON file for the API
+            courses = Course.all.order(updated_at: :desc)
+            render :json => { "items": courses }
+        else
+             redirect_to :root
+        end
+    end
+
+    # Request: GET /courses/review
+    # Authorization: Only available if the requesting user's @current_user.role is Role.admin or Role.moderator.
+    # Unauthorized access: Should get redirected to root (/)
+    # Authorized access: Return a JSON of all courses where course.visibility == Visibility.review in the format { items: [ course1, course2, ...] }
+    def review
+        c_user = current_user()
+        return ( redirect_to :root ) unless logged_in?  # Ensure the user is logged in
+
+        # Verify that they can access this API endpoint
+        if c_user.role == Role.admin or c_user.role == Role.moderator
+
+            # return the courses as a JSON file for the API
+            courses = Course.where(visibility: Visibility.reviewing).order(updated_at: :desc)
+            render :json => { "items": courses }
+        else
+             redirect_to :root
+        end
+    end
+
+    # Request: GET /courses/drafts
+    # Authorization: Only available to logged in user @current_user.id == course.user_id
+    # Unauthorized access: Should get redirected to root (/)
+    # Authorized access: Return a JSON of all courses where course.visibility == Visibility.draft and @current_user.id == course.user_id in the format { items: [ course1, course2, ...] }
+    def drafts
+        c_user = current_user()
+        return ( redirect_to :root ) unless logged_in?  # Ensure the user is logged in
+
+        courses = Course.where(visibility: Visibility.draft, user_id: c_user.id).order(updated_at: :desc)
+        render :json => { "items": courses }
+    end
+
+    # Request: GET /courses/published
+    # Authorization: Available to public
+    # Unauthorized access: Should get redirected to root (/)
+    # Authorized access: Return a JSON of all courses where course.visibility == Visibility.published in the format { items: [ course1, course2, ...] }
+    def published        
+        courses = Course.where(visibility: Visibility.published).order(updated_at: :desc)
+        render :json => { "items": courses }
+    end
+
+    # Request: GET /courses/(:course_id)
+    # Authorization:
+    # - If course.visibility == Visibilityi.publisehd then visible to public
+    # - otherwise only available to admin, moderator, and creator of course
+    # Unauthorized access: Should get redirected to root (/)
+    # Authorized access: Return a JSON of the course where course.id == course_id in the format { item: { title: "How to kill a mockingbird", description : "...", ... } }
+    def getcourse
+        c_user = current_user()
+        return ( redirect_to :root ) unless logged_in?  # Ensure the user is logged in
```

Comment:
This is actually a bug, published courses are visible even when not logged in. The login check is being done a couple more lines down for non-published